### DISCUSSION TITLE
Live traffic recording and export to pcap

### DISCUSSION THREAD

**Discussion by didiksudyana** [nfstream/nfstream#134] [Q&A]
Hi, I am very like with the idea of NFStream in recording the live traffic by detecting a new flow connection in a new packet, then store all of the next packets within the same flow connection. After the flow finish or exceed the timeout, NFStream then will directly export it to csv or pandas. 

However, I have a question. Can we export the finished flow to pcap file, instead of only csv or pandas? Because I need the pcap file for further checking. Or can someone help me to find how to to do it?

Thank you very much.

**Comment by drnpkr** [comment]
Hi @didiksudyana,

NFStream is not suitable for developing a PCAP file. You will need to use WireShark/tShark or tcpdump. These are tools for capturing the traffic trace observed on your network interface card and storing them in a PCAP/PCAPNG format.

I hope this clarifies your questions.

**Reply by didiksudyana** [reply]
Thank you, Prof. Adrian for answering my question.

Yes, Wireshark or tcpdump can be used to export the traffic to a pcap format.
However, those tools are not suitable in our case since they can't save the traffic based on a flow like in NFStream.
In NFStream, the traffic will be grouped to each flow, then everytime a new packet comes, NFStream will check whether this packet belongs to the existing flow or a new flow.

In our case, we have to group all live traffic to each flow, then if the flow is finish, we need to do two things, firstly, we will extract it to pandas and analyze the traffic, then secondly, we also have to extract the finished flow to a pcap file. NFStream is very suitable for us in this case, however, we don't know how to modify the NFStream to extract the finished flow not only to the pandas, but also to a pcap.
